test/web-platform-tests/to-upstream/html/editing/activation/click-event-properties.html

```diff
@@ -104,6 +104,24 @@
   assert_equals(e.offsetY, 22);
 }, "MouseEvent should provide the correct computed values when the dispatch flag is not set");
 
+async_test(t => {
+  const element = document.querySelector("button");
+  element.addEventListener("click", t.step_func_done(e => {
+    // changing these during dispatch should not mess up pageX/Y
+    window.scrollX = 0;
+    window.scrollY = 0;
+    assert_equals(e.pageX, 110);
+    assert_equals(e.pageY, 220);
+    assert_equals(e.offsetX, 0);
+    assert_equals(e.offsetY, 0);
+  }));
+  window.scrollX = 10;
+  window.scrollY = 20;
+  element.dispatchEvent(new MouseEvent("click", {
+    clientX: 100, clientY: 200, view: window
+  }));
```

Comment:
In this test, we should check also the case with:
```js
const ev = new MouseEvent();
window.scrollX = 10;
window.scrollY = 20;
el.dispatchEvent(ev);
```
and expect the updated `window.scroll*` values to be used inside the dispatch.

Maybe there are also WPTs for this case?

Reply:
I'm not sure if that's a valid test in terms of the spec, but I suppose that depends on how you interpret the "when" behind `the position where the event occurred`. Does the event occur when you construct it, or when you dispatch it? Most of the relevant coordinates are set during construction via MouseEventInit, so it seems reasonable to conclude that that's when the coordinate was selected.

That said, I'll take a closer look at the existing WPTs and see what can be done, in particular I'll look to add/enhance tests to coverage what I outlined in my other comment.

One problem I've noticed with my tests is that setting `scrollX/Y` can't be upstreamed, since those properties should be readonly and it would be better to call `scroll(...)`. But jsdom doesn't implement scroll yet 🙃

Reply:
Looks like there aren't many CSSOM View MouseEvent tests, but what I've found bears out my testing, e.g. https://github.com/web-platform-tests/wpt/blob/master/css/cssom-view/mouseEvent.html has a failure for both [WebKit](https://github.com/WebKit/WebKit/blob/136a23f12bae93a6dd2c9f93565aa6da6efdfef5/LayoutTests/imported/w3c/web-platform-tests/css/cssom-view/mouseEvent-expected.txt#L4) and [Chrome](https://github.com/chromium/chromium/blob/main/third_party/blink/web_tests/external/wpt/css/cssom-view/mouseEvent-expected.txt#L3) since they don't consider the scroll offset.

Doesn't look like there's any testing around what happens during dispatch :-/